cli/Cargo.toml

```diff
@@ -118,6 +118,7 @@ rustyline-derive = "=0.7.0"
 serde.workspace = true
 serde_repr.workspace = true
 shell-escape = "=0.1.5"
+sysinfo = "=0.29.10"
```

Comment:
Discussed offline - this is adding another dynamic library on startup. Waiting on @littledivy's review

Reply:
Looking at the usage I think we can manually implement APIs like we did in https://github.com/denoland/deno/pull/16441 

To keep it simple, use sysinfo crate as non-macOS dependency and have a Apple-specific implementation:

https://github.com/GuillaumeGomez/sysinfo/blob/84c3b36a730a91229dd5a738c725aac81a08c7c8/src/unix/apple/macos/process.rs#L663-L692